runtime/bundles/org.eclipse.core.runtime/src/org/eclipse/core/internal/runtime/InternalPlatform.java

```diff
@@ -745,6 +752,13 @@ private DebugOptions getDebugOptions() {
 	}
 
 	private void closeOSGITrackers() {
+		ExtendedLogReaderService logReader = logReaderTracker.getService();
+		logs.values().removeIf(log -> {
```

Comment:
Probably shouldn't even do this stream if `logReader` is null.

A stream with `removeIf` seems like an expensive way to do this.  Why not do this?
```java
ExtendedLogReaderService logReader = logReaderTracker.getService();
if (logReader != null) {
  logs.values().forEach((log) -> logReader.removeLogListener(log));
}
logs.clear();
```

Reply:
> A stream with `removeIf` seems like an expensive way to do this. Why not do this?

Not sure if we really safe anything here:

1. logReader should be not null in almost all cases
2. iterate first and then clear seem not much less "expensive" as we are shut down and there are not millions of bundles having a logger anyways...

Reply:
This is clearly a style discussion.  I think only doing a single null check is better on its own, but I also think the 5 lines of code is more simple and doing a single `clear()` call is more optimal than removing each element individually.

Is it going to make any measurable difference, I doubt seriously it.  But I still think it is better.

Reply:
Done!